lib/twitter_cldr/utils/range_set.rb

```diff
@@ -161,7 +161,72 @@ def difference(range_set)
         union(range_set).subtract(intersection(range_set))
       end
 
-      protected
+      def each_range(&block)
+        ranges.each(&block)
+      end
+
+      def each(&block)
+        if block_given?
+          ranges.each do |range|
+            range.each(&block)
+          end
+        else
+          to_enum(__method__)
+        end
+      end
+
+      private
+
+      def includes_numeric?(num)
+        result = bsearch do |range|
+          if num >= range.first && num <= range.last
+            0
+          elsif num < range.first
+            -1
+          else
+            1
+          end
+        end
+
+        !!result
+      end
+
+      def includes_range?(range)
+        bsearch do |cur_range|
+          if full_overlap?(cur_range, range)
+            0
+          elsif front_overlap?(cur_range, range)
+            return false
```

Comment:
Uh, I got confused for a second here with this mix of -1/0/1 and booleans. How about separating this into an early return and placing it above the condition?

```ruby
bsearch do |cur_range|
  return false if front_overlap?(..) || rear_overlap?(..)

  if full_overlap?(...)
    ...
  end
end
```